### PR TITLE
#194 Only generate field / imports for a referenced mapper if it is actu...

### PR DESCRIPTION
**Pull request by sjaakd**
...ally used

And as bonus, some small template updates.. (fixing empty lines in case of omitted fields).

### CODE REVIEW

File: processor/src/main/java/org/mapstruct/ap/model/DecoratorConstructor.java [hunk 1]
```diff
@@ -29,13 +28,14 @@
  *
  * @author Gunnar Morling
  */
-public class DecoratorConstructor extends ModelElement {
+public class DecoratorConstructor extends Field {
```
Comment (gunnarmorling) [inline processor/src/main/java/org/mapstruct/ap/model/DecoratorConstructor.java]: Constructor extending Field? That seems a bit odd.

File: processor/src/test/java/org/mapstruct/ap/test/references/statics/StaticsTest.java [hunk 2]
```diff
@@ -44,4 +55,19 @@ public void shouldUseStaticMethod() {
         assertThat( result ).isNotNull();
         assertThat( result.getCategory() ).isEqualTo( Category.STRONG ); // why settle for less?
     }
+
+    @Test
+    @WithClasses( {  BeerMapperWithNonUsedMapper.class, NonUsedMapper.class } )
+    public void shouldNotImportNonUsed() {
+
+        Beer beer = new Beer(); // what the heck, open another one..
```
Comment (gunnarmorling) [inline processor/src/test/java/org/mapstruct/ap/test/references/statics/StaticsTest.java]: :-)

**Comment by gunnarmorling** [comment]
Looks good overall, only I find the re-usage of fields for constructors a bit dubious. Should pass them separately to the `Decorator` constructor instead of sneaking them in throught the fields list?

File: processor/src/main/java/org/mapstruct/ap/model/DecoratorConstructor.java (hunk 1 above)
Comment (sjaakd) [inline processor/src/main/java/org/mapstruct/ap/model/DecoratorConstructor.java]: It is.. However its confirming the current practice..(that's why I ran into problems)..  I'll change that.

**Comment by sjaakd** [comment]
> Looks good overall, only I find the re-usage of fields for constructors a bit dubious. Should pass them separately to the Decorator constructor instead of sneaking them in throught the fields list?

Good idea. Of course. 

I'll fix.. Do you mind that I merge immediately after (or do you still want to see it)?

**Comment by gunnarmorling** [comment]
Yes, sure, go ahead. Generally just move forward in such case.

**Pull request merged**